### PR TITLE
Play setSize and swap in a transaction with autoExpand dance included

### PR DESCRIPTION
**Pull request by volfied**
We were playing the `setAutoExpand` calls outside the `playLayerActions` calls, so on an undo, we would get back to the state where we turned auto expand off.

Now we use the handy transactions to make sure the autoExpand dance is included in the history state.

Addresses #3357

### CODE REVIEW

File: src/js/actions/transform.js [hunk 1]
```diff
@@ -504,6 +504,11 @@ define(function (require, exports) {
             headlights.logEvent("edit", "layers", "swap-non-artboard");
         }
 
+        var transaction = descriptor.beginTransaction(options),
```
Comment (mcilroyc) [inline src/js/actions/transform.js]: style nit: it seems a little awkward how `options` is dangling a dozen or so lines above.

File: src/js/actions/transform.js [hunk 2]
```diff
@@ -533,6 +538,9 @@ define(function (require, exports) {
             })
             .then(function () {
                 return this.transfer(layerActions.resetIndex, document);
+            })
+            .then(function () {
```
Comment (mcilroyc) [inline src/js/actions/transform.js]: My gut says that the transaction should be ended prior to calling resetIndex ... but I don't actually know that it matters.

Comment (volfied) [inline src/js/actions/transform.js]: Good catch, it might not matter but it's probably safer to end it in the correct place for future proofing.

**Comment by volfied** [comment]
Thanks @mcilroyc !

**Comment by mcilroyc** [comment]
alrighty

File: src/js/actions/transform.js [hunk 3]
```diff
@@ -532,6 +535,9 @@ define(function (require, exports) {
                 }
             })
             .then(function () {
+                descriptor.endTransaction(transaction);
```
Comment (iwehrman) [inline src/js/actions/transform.js]: Doesn't this return a promise that you need to wait for???

**Pull request merged**